php-sync/sfs-sync.php

```diff
@@ -305,40 +306,32 @@ public function scheduleBatches ($mode, $toSchedule) {
 
 			$dir = $this->config["BATCHDIR"]."/$mode/$node";
 			if (!is_dir ($dir)) {
-				continue;
+				mkdir ($dir, 0777, TRUE);
 			}
-			
-			$hBatchDir = opendir($dir);
-			if ($hBatchDir === FALSE) {
+			if(!is_dir($dir)){
 				syslog(LOG_CRIT, "Cannot opendir $dir, will retry in ".$this->config["FAILTIME"]." seconds");
 				$this->setFailing ($node);
-				$tasks[] = array ();
+				continue;
+			}
+
+			$allBatches = glob($dir . '/*_*.batch');
```

Comment:
I don't rely on that, we can use readdir - the switch was only because of readability & ease of use.
But readdir doesn't sort entries ("The entries are returned in the order in which they are stored by the filesystem") - this doesn't say the really are sorted by the "timestamp" given - or is up the underlying filesystem, which may handle this different.